consensus/src/main/kotlin/maru/consensus/validation/BlockValidator.kt

```diff
@@ -231,14 +232,16 @@ class PrevCommitSealValidator(
 }
 
 class ExecutionPayloadValidator(
-  private val executionLayerClient: ExecutionLayerClient,
+  private val executionLayerManager: ExecutionLayerManager,
 ) : BlockValidator {
   override fun validateBlock(block: BeaconBlock): SafeFuture<Result<Unit, BlockValidationError>> =
-    executionLayerClient.newPayload(block.beaconBlockBody.executionPayload).thenApply { newPayloadResponse ->
+    executionLayerManager.importPayload(block.beaconBlockBody.executionPayload).thenApply { newPayloadStatus ->
```

Comment:
Using `newPayload` was clearer as `importPayload ` is not importing the block, just storing it.

Reply:
True, but it's not a block, it's a payload 🙂. I can change it back to newPayload if it makes it clearer